chia/wallet/notification_store.py

```diff
@@ -0,0 +1,125 @@
+import dataclasses
+import logging
+from typing import List, Optional
+
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.db_wrapper import DBWrapper2
+from chia.util.ints import uint32, uint64
+
+
+@dataclasses.dataclass(frozen=True)
+class Notification:
+    coin_id: bytes32
```

Comment:
should we also include coin's puzzle hash so we can easily filter by type of notifications?

Reply:
The puzzle hash is parameterized by 4 things:
* The notification mod
* The wallet address it was sent to
* The message
* The amount

I don't see how any of those things would be helpful except the message and amount which are already on this class

Reply:
I'm not making any biases in the wallet for what kind of notifications we sync.  I see this as a UI task: delete the notficiations you don't want.  It doesn't prevent any spam because you can always make a notification of a certain format that has a dead URL for example.